### PR TITLE
Fix formatting of logs coming from client-go

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Fix formatting of logs coming from `client-go`.

### Motivation

`client-go` generates logs on standard error that are not following the standard layout.

### Describe your test plan

Check the operator logs.

Before this PR:
```
{"level":"INFO","ts":"2020-12-08T14:01:50Z","logger":"setup","msg":"starting manager"}
I1208 14:01:50.588462       1 leaderelection.go:242] attempting to acquire leader lease  datadog-operator-system/datadog-operator-lock...
{"level":"INFO","ts":"2020-12-08T14:01:50Z","logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
I1208 14:02:08.255290       1 leaderelection.go:252] successfully acquired lease datadog-operator-system/datadog-operator-lock
{"level":"INFO","ts":"2020-12-08T14:02:08Z","logger":"controller","msg":"Starting EventSource","reconcilerGroup":"datadoghq.com","reconcilerKind":"DatadogAgent","controller":"datadogagent","source":"kind source: /, Kind="}
```

With this PR:
```
{"level":"INFO","ts":"2020-12-08T19:49:09Z","logger":"setup","msg":"starting manager"}
{"level":"INFO","ts":"2020-12-08T19:49:09Z","logger":"klog","msg":"attempting to acquire leader lease  datadog-operator-system/datadog-operator-lock...\n"}
{"level":"INFO","ts":"2020-12-08T19:49:09Z","logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
{"level":"INFO","ts":"2020-12-08T19:49:25Z","logger":"klog","msg":"successfully acquired lease datadog-operator-system/datadog-operator-lock\n"}
{"level":"INFO","ts":"2020-12-08T19:49:25Z","logger":"controller","msg":"Starting EventSource","reconcilerGroup":"datadoghq.com","reconcilerKind":"DatadogAgent","controller":"datadogagent","source":"kind source: /, Kind="}
```

### CODE REVIEW

**Review by xornivore — APPROVED** [review]

File: klog.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```
Comment (clamoriniere) [inline klog.go:1]: can we move klog.go in a package `pkg/klog` for example?

**Review by clamoriniere — COMMENTED** [review]
small comment about package organisation

File: klog.go (hunk 1 above)
Comment (L3n41c) [reply klog.go]: Done: https://github.com/DataDog/datadog-operator/pull/192/commits/48d7bf11a9320226e775fe8fbb983697e88c5390
Thanks for the suggestion.

File: main.go [hunk 2]
```diff
@@ -52,6 +53,8 @@ func init() {
 	utilruntime.Must(datadoghqv1alpha1.AddToScheme(scheme))
 	utilruntime.Must(edsdatadoghqv1alpha1.AddToScheme(scheme))
 	// +kubebuilder:scaffold:scheme
+
+	klog.Configure(ctrl.Log.WithName("klog"))
```
Comment (clamoriniere) [inline main.go:57]: Should be put klog configuration first in the init() ? if never the `utilruntime` also log something

Comment (L3n41c) [reply main.go]: Done !

**Review by clamoriniere — APPROVED** [review]
💯

**Pull request merged**